src/parser/monk/brewmaster/modules/spells/azeritetraits/StraightNoChaser.js

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import { Line as LineChart } from 'react-chartjs-2';
+import Analyzer from 'parser/core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import { calculateAzeriteEffects } from 'common/stats';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import TraitStatisticBox from 'interface/others/TraitStatisticBox';
+import SpellLink from 'common/SpellLink';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import { formatNumber, formatPercentage } from 'common/format';
+
+import SpellUsable from '../../core/SpellUsable';
+
+const SNC_PROC_CHANCE = 0.08;
+
+export default class StraightNoChaser extends Analyzer {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+    stats: StatTracker,
+    spellUsable: SpellUsable,
+  };
+
+  armor = 0;
+
+  get isbCasts() {
+    return this.abilityTracker.getAbility(SPELLS.IRONSKIN_BREW.id).casts;
+  }
+
+  get resets() {
+    return this.spellUsable._SNCResets;
+  }
+
+  get uptimePct() {
+    return this.selectedCombatant.getBuffUptime(SPELLS.STRAIGHT_NO_CHASER_BUFF.id) / this.owner.fightDuration;
+  }
+
+  get avgArmor() {
+    return this.armor * this.uptimePct;
+  }
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.STRAIGHT_NO_CHASER.id);
+    if(!this.active) {
+      return;
+    }
+
+    this.armor = this.selectedCombatant.traitsBySpellId[SPELLS.STRAIGHT_NO_CHASER.id].reduce((sum, rank) => {
+      const [armor] = calculateAzeriteEffects(SPELLS.STRAIGHT_NO_CHASER.id, rank);
+      return sum + armor;
+    }, 0);
+
+    this.stats.add(SPELLS.STRAIGHT_NO_CHASER.id, {
+      armor: this.armor,
+    });
+  }
+
+  // currently literal copypasta from the `MasteryValue` module. Once
+  // we're ready to start moving to next this will get replaced with a
+  // generic implementation.
+  plot() {
+    // not the most efficient, but close enough and pretty safe
+    function binom(n, k) {
+      if(k > n) {
+        return null;
+      }
+      if(k === 0) {
+        return 1;
+      }
+
+      return n / k * binom(n-1, k-1);
+    }
+
+    // pmf of the binomial distribution
+    const reset_prob = (i) => binom(this.isbCasts, i) * Math.pow(SNC_PROC_CHANCE, i) * Math.pow(1 - SNC_PROC_CHANCE, this.isbCasts - i);
+
+    const reset_probs = Array.from({length: this.isbCasts}, (_x, i) => {
+      return { x: i, y: reset_prob(i) };
+    });
+
+    const RANGE_THRESHOLD = 0.001;
+    const rangeMin = reset_probs.findIndex(({y}) => y >= RANGE_THRESHOLD);
+    const rangeMax = reset_probs.slice(rangeMin).findIndex(({y}) => y < RANGE_THRESHOLD);
```

Comment:
This is most likely wrong - `slice()` returns a shallow copy of the original array, so `findIndex` doesn't preserve the original indexes (tried in console):
```
const a = [0, 0, 1, 2, 1, 0, 0];
console.log(a.findIndex(x => x > 0.001));
> 2
console.log(a.slice(2).findIndex(x => x < 0.001));
> 3
```
Adding the `rangeMax` to `rangeMin` to get the real `rangeMax` might work though - even in the example, 2 + 3 gives 5 which is the index of the first 0 on the right side.

-----------
Other solution might be to completely filter the low values anyway and pass those into the chart directly?
```javascript
// (make reset_probs let) or add the filter after Array.from()
reset_probs = reset_probs.filter(({ y }) => y > RANGE_THRESHOLD);
```

Reply:
good catch! the log i had pulled up happened to look ok so i didn't see it. should be fixed in a moment.

Reply:
![screenshot from 2019-01-04 16-42-42](https://user-images.githubusercontent.com/4909458/50712500-c5dba100-103f-11e9-9e2c-1b60d640f622.png)
looks like this now